src/Entities/Entity.cpp

```diff
@@ -1835,6 +1702,131 @@ void cEntity::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
 
 
 
+bool cEntity::MoveToWorld(cWorld & a_NewWorld, const Vector3d & a_NewPosition)
+{
+	auto PreviousDimension = GetWorld()->GetDimension();
+	if (OnPreWorldTravel(a_NewWorld))
+	{
+		OnPostWorldTravel(PreviousDimension, a_NewPosition);
+		SetPosition(a_NewPosition);  // Just in case :)
+		return true;
+	}
+	return false;
+}
+
+
+
+
+
+bool cEntity::MoveToWorld(const AString & a_WorldName, const Vector3d & a_NewPosition)
+{
+	auto World = cRoot::Get()->GetWorld(a_WorldName);
+	if (World == nullptr)
+	{
+		LOG("%s: Couldn't find world \"%s\".", __FUNCTION__, a_WorldName.c_str());
+		return false;
+	}
+
+	return MoveToWorld(*World, a_NewPosition);
+}
+
+
+
+
+
+bool cEntity::OnPreWorldTravel(cWorld & a_NewWorld)
+{
+	ASSERT(IsTicking());
+
+	if (GetWorld() == &a_NewWorld)
+	{
+		// Don't move to same world
+		return false;
+	}
+
+	//  Ask the plugins if the player is allowed to changing the world
+	if (cRoot::Get()->GetPluginManager().CallHookEntityChangingWorld(*this, a_NewWorld))
+	{
+		// A Plugin doesn't allow the player to changing the world
+		return false;
+	}
+
+	// Prevent further ticking in this world
+	SetIsTicking(false);
+
+	GetWorld()->QueueTask(
+		[this, &a_NewWorld](cWorld & a_CurrentWorld)
+		{
+			auto Entity = GetParentChunk()->GetAssociatedEntityPtr(*this);
+
+			class Callback : public cEntityCallback
+			{
+			public:
+				Callback(const std::weak_ptr<cPawn> & a_Pawn) :
+					m_Pawn(a_Pawn)
+				{
+				}
+
+				virtual bool Item(cEntity * a_Entity) override
+				{
+					if (!a_Entity->IsMob())
+					{
+						return false;
+					}
+
+					auto Monster = static_cast<cMonster *>(a_Entity);
+					if (std::equal_to<std::weak_ptr<cPawn>>()(Monster->GetWeakTargetPtr(), m_Pawn))
+					{
+						Monster->SetTarget(nullptr);
```

Comment:
- This stops everyone from targeting this entity. I think you need something similar at `TeleportToCoords` and at `cEntity::Destroy `.
 - You got rid of the `StopEveryoneFromTargetingMe` and friends due to safety concerns, and that's understandable. But now entity  destruction costs O(Number of entities in world) rather than the previous O(Number of entities targeting me). But I guess `Stability and readability > Performance`, so I suppose it's fine for now.